packages/main/src/TabContainer.js

```diff
@@ -420,39 +419,111 @@ class TabContainer extends UI5Element {
+			item._posinset = index + 1;
+			item._setsize = arr.length;
+			item._getTabContainerHeaderItemCallback = _ => this.getDomRef().querySelector(`#${item._id}`);
+			item._itemSelectCallback = this._onItemSelect.bind(this);
+			item._getRealDomRef = () => this.getDomRef().querySelector(`*[data-ui5-stable=${item.stableDomRef}]`);
+			item._realTab = this._selectedTab;
+			item._isTopLevelTab = this.items.some(i => i === item);
+			walk(items, tab => {
+				tab._realTab = item._realTab;
+			});
+		});
+	}
+
+	async _onTabStripClick(event) {
 		const tab = getTab(event.target);
 		if (!tab) {
 			return;
 		}
+		event.preventDefault();
+		event.stopPropagation();
+		if (event.target.hasAttribute("ui5-button")) {
+			this._onTabExpandButtonClick(event);
+			return;
+		}
+
+		if (!tab._realTab._hasOwnContent && tab._realTab.tabs.length) {
+			this._overflowItems = [];
+			this._overflowItems = tab._realTab.subTabs;
+			this._addStyleIndent(this._overflowItems, false);
 
+			this.responsivePopover = await this._respPopover();
+			if (this.responsivePopover.opened) {
+				this.responsivePopover.close();
+			} else {
+				this.responsivePopover.showAt(tab);
+			}
+
+			this.selectTab(this._getRootTab(tab._realTab), this._allItemsAndSubItems.indexOf(this._getRootTab(tab._realTab)));
+			return;
+		}
 		this._onHeaderItemSelect(tab);
 	}
 
-	_onTabStripKeyDown(event) {
-		const tab = getTab(event.target);
-		if (!tab) {
-			return;
+	async _onTabExpandButtonClick(event) {
+		event.stopPropagation();
+		event.preventDefault();
+		let button = event.target;
+		let tabInstance = button.tab;
+
+		if (tabInstance) {
+			tabInstance.focus();
 		}
 
-		if (isEnter(event)) {
-			this._onHeaderItemSelect(tab);
+		if ((event.type === "keydown") && !event.target._realTab.isSingleClickArea) {
+			button = event.target.querySelectorAll(".ui5-tab-expand-button")[0];
+			tabInstance = event.target._realTab;
 		}
 
-		// Prevent Scrolling
-		if (isSpace(event)) {
-			event.preventDefault();
+		// if clicked between the expand button and the tab
+		if (!tabInstance) {
+			this._onHeaderItemSelect(button.parentElement);
+			return;
 		}
+
+		this._overflowItems = tabInstance.subTabs;
+		this._addStyleIndent(this._overflowItems, false);
+
+		this.responsivePopover = await this._respPopover();
+		if (this.responsivePopover.opened) {
+			this.responsivePopover.close();
+		} else if (this._getSelectedInPopover().length) {
+			this.responsivePopover.initialFocus = this._getSelectedInPopover()[0].id;
+		} else {
+			this.responsivePopover.initialFocus = this.responsivePopover.content[0].items.filter(item => item.classList.contains("ui5-tab-overflow-item"))[0].id;
+		}
+		this.responsivePopover.showAt(button);
 	}
 
-	_onTabStripKeyUp(event) {
+	_onTabStripKeyDown(event) {
 		const tab = getTab(event.target);
 		if (!tab) {
 			return;
 		}
 
-		if (isSpace(event)) {
-			this._onHeaderItemSelect(tab);
+		event.preventDefault();
+
+		if (isEnter(event) || isSpace(event)) {
```

Comment:
The "space" should be handled in the "key up" event. This will fix the issue with the "single click area" "space" selection.